sympy/functions/elementary/hyperbolic.py

```diff
@@ -473,14 +498,38 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
+        from sympy import pi, cos, sinh
+        arg = self.args[0]
+        if arg.is_real:
+            return True
+
+        re, im = arg.as_real_imag()
+
+        # if denom = 0, tanh(arg) = zoo
+        denom = cos(im)**2 + sinh(re)**2
```

Comment:
The reason I constructed it because SymPy does not simplify expressions like this out of the box.
```
In [6]: ((2*n+1)*pi/2)%(pi/2)
Out[6]: π⋅(n mod 1/2)

In [7]: cos((2*n+1)*pi/2)
Out[7]: 0

In [8]: ((2*n+1)*pi/2)%(pi/2) == pi/2
Out[8]: False
```
I could use `simplify()` for the expression as well. Both ways are good I think.